pkg/scheduler/apis/config/v1alpha1/defaults.go

```diff
@@ -55,22 +58,62 @@ func SetDefaults_KubeSchedulerConfiguration(obj *kubeschedulerconfigv1alpha1.Kub
 		}
 	}
 
-	if host, port, err := net.SplitHostPort(obj.HealthzBindAddress); err == nil {
-		if len(host) == 0 {
-			host = "0.0.0.0"
-		}
-		obj.HealthzBindAddress = net.JoinHostPort(host, port)
+	// For Healthz and Metrics bind addresses, we want to check:
+	// 1. If the value is nil, default to 0.0.0.0 and default scheduler port
+	// 2. If there is a value set, attempt to split it. If it's just a port (ie, ":1234"), default to 0.0.0.0 with that port
+	// 3. If splitting the value fails, check if the value is even a valid IP. If so, use that with the default port.
+	// Otherwise use the default bind address
+	defaultBindAddress := net.JoinHostPort("0.0.0.0", strconv.Itoa(ports.InsecureSchedulerPort))
+	if obj.HealthzBindAddress == nil {
+		obj.HealthzBindAddress = &defaultBindAddress
 	} else {
-		obj.HealthzBindAddress = net.JoinHostPort("0.0.0.0", strconv.Itoa(ports.InsecureSchedulerPort))
+		if host, port, err := net.SplitHostPort(*obj.HealthzBindAddress); err == nil {
+			if len(host) == 0 {
+				host = "0.0.0.0"
+			}
+			hostPort := net.JoinHostPort(host, port)
+			obj.HealthzBindAddress = &hostPort
+		} else {
+			// Something went wrong splitting the host/port, could just be a missing port so check if the
+			// existing value is a valid IP address. If so, use that with the default scheduler port
+			if host := net.ParseIP(*obj.HealthzBindAddress); host != nil {
+				hostPort := net.JoinHostPort(*obj.HealthzBindAddress, strconv.Itoa(ports.InsecureSchedulerPort))
+				obj.HealthzBindAddress = &hostPort
+			} else {
+				obj.HealthzBindAddress = &defaultBindAddress
```

Comment:
this is good for backwards compatibility, but add a TODO that in v1beta1 we'd just want to let this error, rather than stomping it with a default ip+port